### PR TITLE
Whatshap: process truncated input with no ALL chromosome

### PR DESCRIPTION
**Pull request by vladsavelyev**
MultiQC can pretty much summarise the ALL "chromosome" given available information: all reported fields are either sums, or averages across blocks, so it's a relatively easy addition. Fixes https://github.com/ewels/MultiQC/issues/2093

### CODE REVIEW

**Comment by vladsavelyev** [comment]
@multiqc-bot changelog

File: multiqc/modules/whatshap/whatshap.py [hunk 1]
```diff
@@ -91,6 +97,9 @@ def process_data(data):
         header = next(file_content).strip().split("\t")
         header[0] = header[0].lstrip("#")
 
+        if sample == "GM24385_0_1x_full_opt":
+            pass
```
Comment (ewels) [inline multiqc/modules/whatshap/whatshap.py:101]: Debugging?

Comment (vladsavelyev) [reply multiqc/modules/whatshap/whatshap.py]: Fixed!

File: multiqc/modules/whatshap/whatshap.py [hunk 2]
```diff
@@ -103,25 +109,63 @@ def process_data(data):
...
             results = dict()
-            results["ALL"] = defaultdict(int)
+            results[ALL_CHROM] = defaultdict(int)
+
+        # If ALL chromosome is not present, and there are multiple contigs available,
+        # e.g. in case of a truncated input file, we summarize available contigs
+        # as a fictional 'ALL' chromosome.
+        if ALL_CHROM not in results and len(results) > 1:
+            log.warning(
+                f"Could not find the 'ALL' chromosome for {sample}. Make sure that the input is not truncated. Will summarize information for all available chromosomes instead."
```
Comment (ewels) [inline multiqc/modules/whatshap/whatshap.py:130]:
```suggestion
                f"Could not find the 'ALL' chromosome for {sample}. Make sure that the input is not truncated. MultiQC will summarize information for all available chromosomes instead."
```

**Review by ewels — APPROVED** [review]
Ok, nice! This goes beyond what I had envisaged - I was thinking that we would just catch the exception and throw an error saying "file is truncated, skipping". You're being very kind to the end user 😅 

Great stuff! ✅

**Pull request merged**